Corona-Warn-App/src/main/java/de/rki/coronawarnapp/task/TaskState.kt

```diff
@@ -0,0 +1,47 @@
+package de.rki.coronawarnapp.task
+
+import org.joda.time.Instant
+import kotlin.reflect.KClass
+
+interface TaskState {
+    val request: TaskRequest
+    val createdAt: Instant
+    val startedAt: Instant?
+    val completedAt: Instant?
+    val error: Throwable?
+    val result: Task.Result?
+    val executionState: ExecutionState
+    val type: KClass<out Task<*, *>>
+
+    val isFinished: Boolean
+        get() = executionState == ExecutionState.FINISHED
+
+    val isFailed: Boolean
+        get() = isFinished && error != null
+
+    val isSkipped: Boolean
+        get() = isFinished && result == null && error == null
```

Comment:
I find this check too implicit. It's making assumptions about how `toSkippedState` works (i.e. that it's not setting any result or error). Is there any existing enum we can use to explicitly check if the task was skipped?

Reply:
Not more or less implicit than failed tasks have an exception and successful tasks have a result, so if skipped tasks would have neither. In some way we have to make a contract about that, I think doing that in the interface of `TaskState` fits.
More explicitly defining with a boolean value seems more error prone, logicwise.

After a task is finished, it's "finished" state is defined by result+error, the `isSkipped` is just a helper here.

Reply:
I'm merging the PR now, but if this is an issue we can fix this in an extra PR.